sqlglot/parser.py

```diff
@@ -1551,10 +1574,55 @@ def extend_props(temp_props: t.Optional[exp.Properties]) -> None:
             begin=begin,
             end=end,
             clone=clone,
-            start=seq_start,
-            increment=seq_increment,
         )
 
+    def _parse_create_sequence(self, replace, properties) -> exp.Create | exp.Command:
+        exists = self._parse_exists(not_=True)
+        this = self._parse_table(schema=True)
+
+        seq = exp.Create(
+            this=this,
+            exists=exists,
+            replace=replace,
+            kind="SEQUENCE",
+            properties=properties,
+        )
+
+        options = []
+
+        while self._curr:
+            if self._match_text_seq("AS"):
+                seq.set("data_type", self._parse_types())
+            elif self._match_text_seq("INCREMENT"):
+                self._match_text_seq("BY")
+                seq.set("increment", self._parse_term())
+            elif self._match_text_seq("MINVALUE"):
+                seq.set("minvalue", self._parse_term())
+            elif self._match_text_seq("MAXVALUE"):
+                seq.set("maxvalue", self._parse_term())
+            elif self._match(TokenType.START_WITH) or self._match_text_seq("START"):
+                seq.set("start", self._parse_term())
+            elif self._match_text_seq("CACHE"):
+                # T-SQL allows empty CACHE which is initialized dynamically
+                seq.set("cache", self._parse_term() or exp.Var(this=""))
+            elif self._match_text_seq("OWNED", "BY"):
+                owned = None if self._match_text_seq("NONE") else self._parse_column()
+                seq.set("owned", owned)
+            elif self._match_text_seq("COMMENT", "="):
+                seq.set("comment", self._parse_string())
```

Comment:
We shouldn't be duplicating logic here too, we already handle this as a DDL property:

```python
>>> import sqlglot
>>> sqlglot.parse_one("CREATE TABLE z (a INT) COMMENT='x'")
Create(
  this=Schema(
    this=Table(
      this=Identifier(this=z, quoted=False)),
    expressions=[
      ColumnDef(
        this=Identifier(this=a, quoted=False),
        kind=DataType(this=Type.INT, nested=False))]),
  kind=TABLE,
  properties=Properties(
    expressions=[
      SchemaCommentProperty(                                 <--- this one
        this=Literal(this=x, is_string=True))]))
```

I'm wondering whether we can treat these "options" as "properties" instead, so that we can just repeat the same pattern as in other DDLs: `extend_props(self._parse_properties())`. Maybe this will allow us to get rid of the extra logic, like you mention here:

> Future note: It seems that some overlapping options are also parsed in exp.GeneratedAsIdentityColumnConstraint, maybe a refactor between them is favorable?

There is indeed some overlap and either now or in the future we should figure out if we can reduce the duplication.